### PR TITLE
Use arrays instead of vectors in tests

### PR DESCRIPTION
**Pull request by petrochenkov**
Tests often use `vec![1, 2, 3]` instead of shorter and faster `[1, 2, 3]`.
This patch removes a lot of unnecessary `vec!`s. Hopefully, the tests will compile and run a bit faster.

### CODE REVIEW

**Comment by Manishearth** [comment]
Needs rebase.

**Comment by petrochenkov** [comment]
Rebased and removed some more `vec!`s from newer code.

File: src/libcollections/binary_heap.rs [hunk 1]
```diff
@@ -480,7 +480,7 @@ impl<T: Ord> BinaryHeap<T> {
     /// heap.push(3);
     ///
     /// let vec = heap.into_sorted_vec();
-    /// assert_eq!(vec, vec![1, 2, 3, 4, 5, 6, 7]);
```
Comment (Manishearth) [inline src/libcollections/binary_heap.rs]: How will this work?

Comment (Manishearth) [inline src/libcollections/binary_heap.rs]: Hey, it works. Never mind :)

**Comment by petrochenkov** [comment]
It didn't pass local make check, I've fixed the broken test.

**Pull request merged**